pageserver/src/walredo.rs

```diff
@@ -410,7 +409,7 @@ impl PostgresRedoManager {
                 flags,
             } => {
                 // sanity check that this is modifying the correct relation
-                let (rel, blknum) = key_to_rel_block(key).context("invalid record")?;
+                let (rel, blknum) = key_to_rel_block(&key).context("invalid record")?;
```

Comment:
just curious: why change this API to take a reference, esp if it's marked `inline(always)`?

Reply:
+1, IMO it would be best to keep the function's API as it is in this PR, and then, in a later PR, move these simple conversion functions into inherent impls on the `Key` type taking it by reference.

Reply:
To move slightly less data around in the unlikely event of the function not being inlined. But yeah, it shouldn't matter. I'll drop that commit.

Reply:
Done in https://github.com/neondatabase/neon/pull/6406/commits/b58ec702cbfc090b09e767f4e302df124bb194f0